### ISSUE TITLE
prometheus/grafana monitoring setup not destroying load balancer on destruction correctly

### ISSUE THREAD

**Issue by woodcockr** [opendatacube/datacube-k8s-eks#45]
On destroy terraform will fail to destroy the Certificate because it is still in use by the prometheus load balancer - despite the fact that the apps have all been destroyed.

My guess - Looks like a dependency is missing and terraform is destroying things before external-dns and ingress complete the removal of the loadbalancer.

**Comment by tom-butler** [comment]
The load balancer is managed by aws alb ingress controller, not terraform. We could get around this by adding a local-exec to delete the ingress when deleting the helm chart.

**Comment by woodcockr** [comment]
My jhub setup removes the ingress and everything associated with it when I remove the helm chart - aka alb ingress controller doing its job.

I've not looked at this, just repeated it twice, and something is wrong. My guess was that terraform is able to destroy either external-dns or the alb-ingress-controller before the monitoring helm chart is executed. Thus when the monitoring helm chart is run the additional clean up isn't done.

**Comment by woodcockr** [comment]
looks like adding to the `helm_release.prometheus_operator` two more dependencies cures the problem `helm_release.external-dns` and `helm_release.alb-ingress`

**Comment by woodcockr** [comment]
Hmm, maybe not, still coming up on this last run. both the load balancer and the dns entries for mgmt.domainname are persisting after external-dns has been removed so the cert cannot be destroyed.

**Comment by woodcockr** [comment]
I've had "opportunity" to confirm the prometheus hiccups we've experienced during apply/destroy/update cycles. It is as I had guessed previously a case of alb-ingress (in particular) and external-dns not having opportunity to complete their update cycle before they are destroyed.
1. Add delay upon completion of the prometheus destroy - given the dependencies we just need to make this long enough for the others to complete their update cycle
2. Move prometheus into an `apps` directory and place all `apps` the depend on alb-ingress and external-dns updating on 1pps1 destroy to this directory. We can then add ONE delay for the update cycle (rather than one per `app`).
